src/pages/search/advanced_search/FundingSourceFilter.tsx

```diff
@@ -0,0 +1,63 @@
+import { Autocomplete } from '@mui/material';
+import { useQuery } from '@tanstack/react-query';
+import { useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router-dom';
+import { fetchFundingSources } from '../../../api/cristinApi';
+import { ResultParam } from '../../../api/searchApi';
+import { AutocompleteTextField } from '../../../components/AutocompleteTextField';
+import { FundingSource } from '../../../types/project.types';
+import { getLanguageString } from '../../../utils/translation-helpers';
+
+export const FundingSourceFilter = () => {
+  const { t } = useTranslation();
+  const history = useHistory();
+  const searchParams = new URLSearchParams(history.location.search);
+  const currentSearchTerm = searchParams.get(ResultParam.FundingSource);
+
+  const fundingSourcesQuery = useQuery({
+    queryKey: ['fundingSources'],
+    queryFn: fetchFundingSources,
+    meta: { errorMessage: t('feedback.error.get_funding_sources') },
+    staleTime: Infinity,
+    cacheTime: 1_800_000, // 30 minutes
+  });
+  const fundingSourcesList = fundingSourcesQuery.data?.sources ?? [];
+
+  const handleChange = (selectedValue: FundingSource | null) => {
+    if (selectedValue) {
+      searchParams.set(ResultParam.FundingSource, selectedValue.identifier);
+    } else {
+      searchParams.delete(ResultParam.FundingSource);
+    }
+
+    history.push({ search: searchParams.toString() });
+  };
+
+  return (
+    <Autocomplete
+      sx={{ minWidth: '15rem' }}
+      disabled={!fundingSourcesQuery.data}
+      value={fundingSourcesList.find((source) => source.identifier === currentSearchTerm) || null}
```

Comment:
lå [her](https://github.com/BIBSYSDEV/NVA-Frontend/pull/5164#discussion_r1489704790) også
```suggestion
      value={fundingSourcesList.find((source) => source.identifier === currentSearchTerm)}
```

Reply:
Får denne i konsollen om den fjernes:
<img width="851" alt="Screenshot 2024-02-15 at 13 47 13" src="https://github.com/BIBSYSDEV/NVA-Frontend/assets/53341294/0dde4d7a-67ad-4d8c-a25d-176ee058b559">

Reply:
ah stemmer det, tenkte ikke på at det var value :)
Bare pass på at det ikke blir noe kluss med `||` der da, siden det også kan evalueres til `true`/`false`, mens vi ofte bruker `??` i sånne tilfeller, som bare tar verdien til høyre om den til venstre er `undefined` eller `null`